pkg/eval/str/str.go

```diff
@@ -8,6 +8,316 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare returns an integer comparing two strings lexicographically. The
```

Comment:
* Go's convention of using the symbol name (`compare`) here unfortunately doesn't work well for Elvish; In Go all exported symbols are capitalized and can be used as the first word of an English sentence. In Elvish all symbols are (for now at least) and they are not capitalized.

  Existing docs start the sentence with the verb, so just start the sentence with "Returns" here (but see bullets below on this particular one).

  Same elsewhere.

* Elvish does not have return values, say "outputs" instead of "returns".

  Same elsewhere.

* I know you copied Go's doc here, but the original wording can use some improvement. An "integer comparing two strings" is not quite intuitive. I'd rather this say:

  > Compares two strings and outputs an integer representing the lexicographical ordering of the two strings. The result...

Reply:
Done